src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/service/ExternalTaskWorker.java

```diff
@@ -48,20 +52,49 @@ public void setupClient() {
     }
 
     public void checkIdempotency(ExternalTask externalTask, ExternalTaskService externalTaskService) {
-        Boolean isDuplicate =  externalTask.getVariable("isDuplicate");
-        if (isDuplicate == null || isDuplicate) {
-            Map<String, Object> processVariables = singletonMap(
-                "isDuplicate",
-                false
-            );
+        IdempotentId idempotentId = getIdempotentId(externalTask);
 
-            LOGGER.info("Duplicate was hit.");
+        Optional<IdempotentKeys> idempotentRow = idempotentKeysRepository.findById(idempotentId);
 
-            externalTaskService.complete(externalTask, processVariables);
-        } else {
-            externalTaskService.complete(externalTask);
-            LOGGER.info("Duplicate was true");
+        idempotentRow.ifPresentOrElse(
+            (row) -> handleIdempotentIdIsPresentInDb(externalTask, externalTaskService, row),
+            () -> handleIdempotentIdIsNotPresentInDb(externalTask, externalTaskService, idempotentId)
+        );
+    }
 
+    private IdempotentId getIdempotentId(ExternalTask externalTask) {
+        String idempotentKey = externalTask.getVariable("idempotentKey");
+        String tenantId = externalTask.getVariable("jurisdiction");
```

Comment:
We probably want this to come in as a tenantId.  I'd expect the incoming message to also have a tenantId during initiation?

Reply:
I think we said that for the moment to use the jurisdiction but yeah if there is a tenantId it's better to use that.